### DISCUSSION TITLE
Authentication service configured with Okta IdP saves empty username, name and email

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#482] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12.3.2

### Description

We have configured the authentication-service to use Okta as IdP.
The `authentication-service` requires the `users` CRUD collection to store some information about the users.
We noticed that the fields `email`, `username` and `name` are always set to empty string "" and we don't understand why.
Maybe the Okta `userinfo` endpoint doesn't provide those information, but we don't know, and we don't know how to test this hypothesis.

Have you ever encountered this issue? 

Thanks,

Luca

**Comment by theliuk** [comment]
We tried to remove the fields `name`, `username` and `email` from the collection definition and the application doesn't work anymore, because the CRUD answers 400. This means that the authentication service tries to save the fields as empty strings.

**Reply by malta895** [reply]
Yes I can confirm that non-existing values in the provider `/userinfo` response fall back to the empty string, without raising any errors in case they are missing.

**Comment by malta895** [comment, marked as answer]
Hi @theliuk,

> Maybe the Okta userinfo endpoint doesn't provide those information, but we don't know, and we don't know how to test this hypothesis.

This is very likely what is happening, the authentication service takes those fields from the provider's userinfo response.

Can you make sure that the provider is correctly configured to return those fields from its `/userinfo` endpoint?

**Reply by theliuk** [reply]
Hi @malta895,
We have to ask the devops team to check.

Thank you
